drivers/gpu/drm/vc4/vc4_hdmi.c

```diff
@@ -1185,10 +1187,11 @@ static void vc4_hdmi_encoder_pre_crtc_configure(struct drm_encoder *encoder,
 	 * FIXME: When the pixel freq is 594MHz (4k60), this needs to be setup
 	 * at 300MHz.
 	 */
-	ret = clk_set_min_rate(vc4_hdmi->pixel_bvb_clock,
-			       (hsm_rate > VC4_HSM_MID_CLOCK ? 150000000 : 75000000));
-	if (ret) {
-		DRM_ERROR("Failed to set pixel bvb clock rate: %d\n", ret);
+	vc4_hdmi->bvb_req = clk_request_start(vc4_hdmi->pixel_bvb_clock,
+					      (hsm_rate > VC4_HSM_MID_CLOCK ? 150000000 : 75000000));
+	if (IS_ERR(vc4_hdmi->bvb_req)) {
```

Comment:
This breaks Pi3 which doesn't have a vc4_hdmi->pixel_bvb_clock and so this returns `-EINV`
The other clock function (`clk_set_rate`/`clk_set_min_rate`) return `0` (success) when called with a null clock. I wonder if this should?
e.g. `clk_request_start` returns null if `clk` is null, and clk_request_done does nothing with a null `clkreq`. 
Protecting this with a null check would also work (but is an extra thing that may be unexpected when converting from `clk_set_min_rate`).